libcloud/test/storage/test_s3.py

```diff
@@ -905,6 +907,29 @@ def test_upload_big_object_via_stream(self):
         self.assertEqual(obj.name, object_name)
         self.assertEqual(obj.size, CHUNK_SIZE * 3)
 
+    def test_upload_object_via_stream_guess_file_mime_type(self):
+        def dummy_content_type(name):
+            return 'application/zip', None
+
+        old_func = libcloud.utils.files.guess_file_mime_type
+        libcloud.utils.files.guess_file_mime_type = dummy_content_type
+
+        if self.driver.supports_s3_multipart_upload:
+            self.mock_response_klass.type = 'MULTIPART'
+        else:
+            self.mock_response_klass.type = None
+
+        container = Container(name='foo_bar_container', extra={},
+                              driver=self.driver)
+        object_name = 'foo_test_stream_data'
+        iterator = BytesIO(b('234'))
+        try:
+            self.driver.upload_object_via_stream(container=container,
+                                                 object_name=object_name,
+                                                 iterator=iterator)
```

Comment:
non blocking nitpick: Is there a way to check that the correct content type was used during the upload?

Reply:
I tried, but `upload_object_via_stream` don't return anything useful to validate it. I'd have to mock `_initiate_multipart` to assert headers or extract all header generation to another method.

Reply:
I changed this test to use a mock. I added a check to assert if my mock function is called. Not perfect, but at least we know the code is trying to guess the mimetype.

Reply:
Great, thanks! Another non-blocking nitpick that really you don't have to follow: can we try to be more resilient to future typos here? See https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html. I think using autospec would work in this case?

Reply:
Great article. I had mock assert issues in the past, I know how it is painful to debug. Added `autospec=True` and it works fine. :+1: